geonode/documents/views.py

```diff
@@ -200,9 +200,6 @@ def form_valid(self, form):
             )
 
         self.object.handle_moderated_uploads()
-        resource_manager.set_permissions(
-            None, instance=self.object, permissions=form.cleaned_data["permissions"], created=True
-        )
 
```

Comment:
i would keep here also the `resource.set_default_permissions(owner=self.request.user)` we cannot assume that the update workflow will always have the permissions set. If for any reason is going to be removed, we will lose again the permissions